lib/mix/lib/mix/tasks/test.ex

```diff
@@ -229,9 +229,18 @@ defmodule Mix.Tasks.Test do
   end
 
   defp parse_files([], test_paths) do
+    warn_about_incorrect_extensions_in(test_paths)
+
     test_paths
   end
 
+  defp warn_about_incorrect_extensions_in(test_paths) do
+    for file <- Mix.Utils.extract_files(test_paths, "*_test.ex") do
+      IO.puts "Warning: #{file} has incorrect extension and won't be run. " <>
```

Comment:
@paulcsmith `Mix.shell.(info|error)` is preferable in almost every situation inside mix tasks. The main difference is that `Mix.shell` can be replaced, might be useful for mix-only output handling. Another thing is that default `Mix.Shell.IO` will print application in which `Mix.shell.(info|error)` call happens, important for umbrella applications.
There are some relevant commits for it from last 2 weeks: https://github.com/elixir-lang/elixir/commit/f858ccf205302f14abd14536c106e97a248a79cf, https://github.com/elixir-lang/elixir/commit/21ff94de1f5df137f8bbb061ab0d6979831daac8.

Sorry for late reply, was away from the computer last 2 days. :smile: